src/GSoft.ComponentModel.DataAnnotations/GuidAttribute.cs

```diff
@@ -7,31 +7,52 @@ namespace GSoft.ComponentModel.DataAnnotations;
 [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter)]
 public sealed class GuidAttribute : ValidationAttribute
 {
-    internal const string ErrorMessageWithoutFormatFormat = "The {0} field must be a well-formatted GUID";
-    internal const string ErrorMessageWithFormatFormat = "The {{0}} field must be a well-formatted GUID of format '{0}'";
+    internal const string ErrorMessageNonEmptyPart = " non-empty";
+    internal const string ErrorMessageWithoutFormatFormat = "The {0} field must be a well-formatted{1} GUID";
+    internal const string ErrorMessageWithFormatFormat = "The {0} field must be a well-formatted{1} GUID of format '{2}'";
 
-    public GuidAttribute(string? format = null)
-        : base(() => ErrorMessageFormatAccessor(format))
+    public GuidAttribute()
+    {
+        this.Format = null;
+        this.AllowEmpty = true;
+    }
+
+    public GuidAttribute(string format)
     {
         this.Format = format;
+        this.AllowEmpty = true;
     }
 
     public string? Format { get; }
 
-    private static string ErrorMessageFormatAccessor(string? format)
-    {
-        return format == null ? ErrorMessageWithoutFormatFormat : string.Format(CultureInfo.InvariantCulture, ErrorMessageWithFormatFormat, format);
-    }
+    public bool AllowEmpty { get; set; }
 
     public override bool IsValid(object? value) => value switch
     {
         null => true,
+        Guid valueAsGuid => this.IsValidGuid(valueAsGuid),
         string valueAsString => this.IsValidGuid(valueAsString),
         _ => false,
     };
 
     private bool IsValidGuid(string valueAsString)
     {
-        return this.Format is null ? Guid.TryParse(valueAsString, out _) : Guid.TryParseExact(valueAsString, this.Format, out _);
+        Guid guid;
+        return this.Format is null
+            ? Guid.TryParse(valueAsString, out guid) && this.IsValidGuid(guid)
+            : Guid.TryParseExact(valueAsString, this.Format, out guid) && this.IsValidGuid(guid);
+    }
+
+    private bool IsValidGuid(Guid guid)
+    {
+        return this.AllowEmpty || guid != Guid.Empty;
+    }
+
+    public override string FormatErrorMessage(string name)
```

Comment:
Recently discovered this virtual method which is better IMO than defining a static error message handler accessor at the constructor level